### PR TITLE
Corrected compiler error, inaccurate information in passing array topic

### LINKED ISSUE #6325: First example (ArrayClass)
This line does not compile:
arr = (arr.Reverse()).ToArray();
I could write "Array.Reverse(arr);" instead but then the array is modified even outside the method.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a887e08e-64e4-9169-42b3-1cdbf829316f
* Version Independent ID: 0d58869d-c6a9-8a7b-d345-df56b0873ca6
* Content: [Passing Arrays as Arguments (C# Programming Guide)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/arrays/passing-arrays-as-arguments)
* Content Source: [docs/csharp/programming-guide/arrays/passing-arrays-as-arguments.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/arrays/passing-arrays-as-arguments.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by rpetrusha**
## Corrected compiler error, inaccurate information in passing array topic

Fixes #6325

### CODE REVIEW

File: docs/csharp/programming-guide/arrays/codesnippet/CSharp/passing-arrays-as-arguments_4.cs [hunk 1]
```diff
@@ -1,68 +1,57 @@
...
+
+class ArrayExample
 {
     static void PrintArray(string[] arr)
     {
         for (int i = 0; i < arr.Length; i++)
         {
-            System.Console.Write(arr[i] + "{0}", i < arr.Length - 1 ? " " : "");
+             Console.Write(arr[i] + "{0}", i < arr.Length - 1 ? " " : "");
         }
```
Comment (pkulikov) [inline docs/csharp/programming-guide/arrays/codesnippet/CSharp/passing-arrays-as-arguments_4.cs]: The whole `for` loop can be replaced with the call to [string.Join](https://docs.microsoft.com/en-us/dotnet/api/system.string.join?view=netcore-2.1#System_String_Join_System_String_System_String___). That makes the method one-liner:
```csharp
static void PrintArray(string[] arr) => Console.WriteLine(string.Join(" ", arr));
```

File: docs/csharp/programming-guide/arrays/codesnippet/CSharp/passing-arrays-as-arguments_4.cs [hunk 2]
```diff
@@ -1,68 +1,57 @@
...
     }
 
     static void ChangeArray(string[] arr)
     {
-        // The following attempt to reverse the array does not persist when
-        // the method returns, because arr is a value parameter.
-        arr = (arr.Reverse()).ToArray();
-        // The following statement displays Sat as the first element in the array.
-        System.Console.WriteLine("arr[0] is {0} in ChangeArray.", arr[0]);
+        // // Reversing the array persists when the method returns.
```
Comment (pkulikov) [inline docs/csharp/programming-guide/arrays/codesnippet/CSharp/passing-arrays-as-arguments_4.cs]: nit: too many `//`
Also consider rephrasing: *Reversing the array persists...* -> *The array stays reversed...*

File: docs/csharp/programming-guide/arrays/codesnippet/CSharp/passing-arrays-as-arguments_4.cs [hunk 3]
```diff
@@ -1,68 +1,57 @@
...
     }
 
     static void Main()
     {
         // Declare and initialize an array.
         string[] weekDays = { "Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat" };
-
-        // Pass the array as an argument to PrintArray.
+        // Display the array elements.
         PrintArray(weekDays);
```
Comment (pkulikov) [inline docs/csharp/programming-guide/arrays/codesnippet/CSharp/passing-arrays-as-arguments_4.cs]: nit: `PrintArray` -> `DisplayArray` and the comment above the call can be removed.

File: docs/csharp/programming-guide/arrays/passing-arrays-as-arguments.md [hunk 4]
```diff
@@ -11,48 +11,48 @@ Arrays can be passed as arguments to method parameters. Because arrays are refer
...
   
- [!code-csharp[CsProgGuideArrays#35](../../../csharp/programming-guide/arrays/codesnippet/CSharp/passing-arrays-as-arguments_3.cs)]  
+ [!code-csharp[CsProgGuideArrays#35](codesnippet/CSharp/passing-arrays-as-arguments_3.cs)]  
   
 ## Example  
   
 ### Description  
- In the following example, an array of strings is initialized and passed as an argument to a `PrintArray` method for strings. The method displays the elements of the array. Next, methods `ChangeArray` and `ChangeArrayElement` are called to demonstrate that sending an array argument by value does not prevent changes to the array elements.  
+ In the following example, an array of strings is initialized and passed as an argument to a `PrintArray` method for strings. The method displays the elements of the array. Next, the `ChangeArray` method is called to reverse the array elements, and then the `ChangeArrayElement` method is called to modify the first three elements of the array. After each method returns, the `PrintArray` method is called to show that passing an array by value does not prevent changes to the array elements.
```
Comment (pkulikov) [inline docs/csharp/programming-guide/arrays/passing-arrays-as-arguments.md]: First, if you rename `PrintArray` to `DisplayArray`, don't forget to make it here as well.
Second, there is no `ChangeArrayElement` method, plural is used in the code: `ChangeArrayElements`

**Comment by rpetrusha** [comment]
Closing and reopening to begin new build.

**Comment by rpetrusha** [comment]
Thank you so much for the thorough and careful review, @pkulikov! I've addressed all your comments.

**Comment by rpetrusha** [comment]
Closing and opening to restart build.

File: docs/csharp/programming-guide/arrays/passing-arrays-as-arguments.md [hunk 5]
```diff
@@ -11,48 +11,48 @@ Arrays can be passed as arguments to method parameters. Because arrays are refer
...
   
- [!code-csharp[CsProgGuideArrays#35](../../../csharp/programming-guide/arrays/codesnippet/CSharp/passing-arrays-as-arguments_3.cs)]  
+ [!code-csharp[CsProgGuideArrays#35](codesnippet/CSharp/passing-arrays-as-arguments_3.cs)]  
   
 ## Example  
   
 ### Description  
- In the following example, an array of strings is initialized and passed as an argument to a `PrintArray` method for strings. The method displays the elements of the array. Next, methods `ChangeArray` and `ChangeArrayElement` are called to demonstrate that sending an array argument by value does not prevent changes to the array elements.  
+ In the following example, an array of strings is initialized and passed as an argument to a `DisplayArray` method for strings. The method displays the elements of the array. Next, the `ChangeArray` method is called to reverse the array elements, and then the `ChangeArrayElements` method is called to modify the first three elements of the array. After each method returns, the `PrintArray` method is called to show that passing an array by value does not prevent changes to the array elements.
```
Comment (pkulikov) [inline docs/csharp/programming-guide/arrays/passing-arrays-as-arguments.md]: Typo in the last sentence: `PrintArray` -> `DisplayArray`
nit: @rpetrusha how do you like converting the last two sentences into the active voice?
> Next, the `ChangeArray` method reverses the array, and then the `ChangeArrayElements` method updates the first three elements of the array. After each method returns, the `DisplayArray` method shows that passing an array by value doesn't prevent changes to the array.

Comment (BillWagner) [reply docs/csharp/programming-guide/arrays/passing-arrays-as-arguments.md]: typo: in the last sentence, the method should be `DisplayArray`, not `PrintArray`.

**Review by BillWagner — APPROVED** [review]
This is much more clear, @rpetrusha I approved this PR, but I did notice one location where you are still using the previous method name of `PrintArray` instead of `DisplayArray`. Once you make that change, you can :shipit:

File: docs/csharp/programming-guide/arrays/passing-arrays-as-arguments.md [hunk 6]
```diff
@@ -11,48 +11,48 @@ Arrays can be passed as arguments to method parameters. Because arrays are refer
...
  You can initialize and pass a new array in one step, as is shown in the following example.  
   
- [!code-csharp[CsProgGuideArrays#35](../../../csharp/programming-guide/arrays/codesnippet/CSharp/passing-arrays-as-arguments_3.cs)]  
+ [!code-csharp[CsProgGuideArrays#35](codesnippet/CSharp/passing-arrays-as-arguments_3.cs)]  
   
 ## Example  
   
 ### Description  
- In the following example, an array of strings is initialized and passed as an argument to a `PrintArray` method for strings. The method displays the elements of the array. Next, methods `ChangeArray` and `ChangeArrayElement` are called to demonstrate that sending an array argument by value does not prevent changes to the array elements.  
+ In the following example, an array of strings is initialized and passed as an argument to a `DisplayArray` method for strings. The method displays the elements of the array. Next, the `ChangeArray` method reverses the array elements, and then the `ChangeArrayElements` method to modifies the first three elements of the array. After each method returns, the `DisplayArray` method shows that passing an array by value does not prevent changes to the array elements.
```
Comment (pkulikov) [inline docs/csharp/programming-guide/arrays/passing-arrays-as-arguments.md]: @rpetrusha the last nit: please remove not necessary "to" in *...the `ChangeArrayElements` method to modifies the first three elements...*

**Comment by rpetrusha** [comment]
Thanks for taking a final look at the topic, @pkulikov. I've removed the unnecessary "to".

**Pull request merged**